client/app/services/query-string.js

```diff
@@ -0,0 +1,11 @@
+import qs from 'qs';
+
+const parse = queryString => qs.parse(queryString, { allowDots: true, ignoreQueryPrefix: true });
+const onlyParameters = ([k]) => k.startsWith('p_');
+const removePrefix = ([k, v]) => [k.slice(2), v];
+const toObject = (obj, [k, v]) => Object.assign(obj, { [k]: v });
+
+export default () => Object.entries(parse(location.search))
```

Comment:
Since this parsing function is really specific to redash query parameters, I would:
```js
export default parseLocationSearch = () => Object.entries(parse(location.search));

export parseRedashQueryParameters = () => parseLocationSearch()
  .filter(onlyParameters)
  .map(removePrefix)
  .reduce(toObject, {});
```

Wdyt?

Reply:
I think that most of the (non-lib) code in the project is Redash specific but we don't explicitly add "Redash" to function and class names, so I tend to disagree.

Reply:
While I agree that prefixing stuff with "Redash" in our codebase doesn't make sense, @ranbena  has a point that this is not some generic query string parsing function but a specific one for parameters. It makes sense to name it accordingly, like "parametersFromQueryString" or something along this lines.